std/datetime.d

```diff
@@ -1075,6 +1075,25 @@ public:
         //assert(ist.opCmp(ist) == 0);
     }
 
+    /**
+     * Returns: A hash of the $(LREF SysTime)
+     */
+    size_t toHash() const @nogc pure nothrow @safe
+    {
+        if (_stdTime < 0)
+            return (cast(size_t) (_stdTime * -1)) + long.max;
```

Comment:
`_stdTime` is a long, so anything that's `< 0` flip the sign, convert to a `size_t`, and add long.max to avoid collisions with other `_stdTime` values